packages/docs-page/index.tsx

```diff
@@ -57,6 +57,9 @@ export const DocsPageWrapper: FunctionComponent<DocsPageWrapperProps> = ({
   const { asPath } = useRouter()
   const versionInPath = getVersionFromPath(asPath)
 
+  const versionIsLatest =
+    versionInPath && versions?.[0]?.label.includes(versionInPath)
+
```

Comment:
💭 This logic is pretty brittle...

Reply:
This ticket will update https://app.asana.com/0/1100423001970639/1201619796458626/f all products to have a generic (x-patch) latest version

Reply:
Almost feels like this would be made easier with some kind of "activeVersion" data available from a context or something similar. That would also help persisting the version across different sections.

Reply:
@BRKalow Yea, i've been thinking about that, and I'm pretty on board!

I see the `activeVersion` context helping to de-dupe client-side state-wrangling and props drilling.
- 💭 it would still be derived from the `next/router` right (I still understand that as being a more comprehensive source of truth)

I'd also throw in _`activeLane`_ in there (groupings of _subPaths_ — docs/api | plugins — that should be versioned together in the UI)

---

Tangent to your comment, but related to this PR. I just had an idea to return _more_ data from server code
- https://github.com/hashicorp/react-components/blob/main/packages/docs-page/server/loaders/remote-content.ts#L57-L76

The current implementation is kind of causing this client-side [pain point / fragility](https://github.com/hashicorp/react-components/pull/484/files/6e2febb0c696a3604a9edcf545e8533c5fc64a46#diff-3a175ef635fbea4834759b73520c2e10cbef9d293bc962b8860e85aed42b10c5R60-R61)

Reply:
Returning an `isLatest` attribute from the server would be a good starting point for sure. Yeah ultimately we would need to derive the `activeVersion` from the router I think, but like you said avoid prop drilling and de-duping state seems like a win. We can definitely defer it, or maybe use the thoughts here to draft up a more comprehensive task for future us.

Reply:
I'm going to address the server code bit in this PR to unblock consul versioned docs

But for the context bit, I was thinking it'd make sense to do async w/ the consul/packer versioned docs rollout

Reply:
> We can definitely defer it, or maybe use the thoughts here to draft up a more comprehensive task for future us.

I'll make a ticket right now!